includes/class-amp-theme-support.php

```diff
@@ -829,7 +828,21 @@ protected static function ensure_required_markup( DOMDocument $dom ) {
 			) );
 			$head->insertBefore( $meta_viewport, $meta_charset->nextSibling );
 		}
-
+		// Prevent schema.org duplicates.
+		$schema_org_script = null;
+		foreach ( $head->getElementsByTagName( 'script' ) as $script ) {
+			if ( 'application/ld+json' === $script->getAttribute( 'type' ) && preg_match( '/{"@context":"https?:[\\\]\/[\\\]\/schema\.org/', $script->nodeValue ) ) {
+				$schema_org_script = $script->nodeValue;
```

Comment:
Why can't this do `json_decode( $script->nodeValue, false )` and check if `wp_parse_url( $script['@context'], PHP_URL_HOST ) === 'schema.org`?

Reply:
Hi @westonruter, 

Trying to use `wp_parse_url` with `json_decode( $script->nodeValue, false )` breaks the site, but it works fine when nodeValue is converted to an array. The logic however stops working.

I've found that if you comment out these lines in `class-amp-dom-utils.php`, the logic works again:

		$placeholders = self::get_mustache_tag_placeholders();
		$document     = str_replace(
			array_keys( $placeholders ),
			array_values( $placeholders ),
			$document
		);

I opted to keep your previous suggestions since i don't know how important are those lines.

Reply:
OK, I think this makes sense. If I have some JSON that looks like:

```php
{"foo": {"bar":}}
```

The last `}}` there is going to get replaced with a placeholder (since this token is returned by `get_mustache_tag_placeholders`) before parsing the DOM, to then get put back in `AMP_DOM_Utils::get_content_from_dom_node()`. That means we cannot parse JSON from the parsed DOM currently, which is not ideal. To fix this we'd need to limit the replacements to only happen inside of `href` and `src` attributes only, using something like this:

```diff
diff --git a/includes/utils/class-amp-dom-utils.php b/includes/utils/class-amp-dom-utils.php
index 5288c04..f20fba1 100644
--- a/includes/utils/class-amp-dom-utils.php
+++ b/includes/utils/class-amp-dom-utils.php
@@ -77,9 +77,15 @@ class AMP_DOM_Utils {
 		 * elements since it is faster and wouldn't change the outcome.
 		 */
 		$placeholders = self::get_mustache_tag_placeholders();
-		$document     = str_replace(
-			array_keys( $placeholders ),
-			array_values( $placeholders ),
+		$document     = preg_replace_callback(
+			'#\s(src|href)=("[^"]+"|\'[^\']+\')#s',
+			function( $matches ) use ( $placeholders ) {
+				return str_replace(
+					array_keys( $placeholders ),
+					array_values( $placeholders ),
+					$matches[0]
+				);
+			},
 			$document
 		);
```

That would probably be something good to add here to ensure we can read JSON out of the DOM.

Reply:
I've merged the fix for this in #1019 via https://github.com/Automattic/amp-wp/pull/1019/commits/4a10ee479650016e1baa80c1f5c6dbe566247bf5